coalib/files/Fileproxy.py

```diff
@@ -0,0 +1,97 @@
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq("filename")
+class Fileproxy:
+    """
+    The file proxy is a wrapper object to wrap every file. The file proxy is an
+    object that contains properties about the file currently processed (such as
+    the filename). The fileproxy object also protects your file from being
```

Comment:
Also, as this is technical documentation, rather than "protects your file" , state something like " returns only immutable objects for bears to read only.".  Saying `immutable` is key for callers to understand what it does.